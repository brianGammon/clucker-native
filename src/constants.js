export const APP_VERSION = '0.0.0-dev';
export const PRIVACY_POLICY_URI = 'https://cluckerapp.com';
export const TERMS_OF_USE_URI = 'https://cluckerapp.com';
export const BULK_ENTRY_KEY = 'bulkentry';
export const UNNAMED_HEN_KEY = 'unnamed';
export const UNNAMED_HEN_LABEL = 'Unnamed Hen';
