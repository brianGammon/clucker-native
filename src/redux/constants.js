// @flow
export const actionTypes = {
  CREATE_REQUESTED: 'CREATE_REQUESTED',
  CREATE_FULFILLED: 'CREATE_FULFILLED',
  CREATE_REJECTED: 'CREATE_REJECTED',

  UPDATE_REQUESTED: 'UPDATE_REQUESTED',
  UPDATE_FULFILLED: 'UPDATE_FULFILLED',
  UPDATE_REJECTED: 'UPDATE_REJECTED',

  REMOVE_REQUESTED: 'REMOVE_REQUESTED',
  REMOVE_FULFILLED: 'REMOVE_FULFILLED',
  REMOVE_REJECTED: 'REMOVE_REJECTED',

  LISTEN_REQUESTED: 'LISTEN_REQUESTED',
  LISTEN_FULFILLED: 'LISTEN_FULFILLED',
  LISTEN_REJECTED: 'LISTEN_REJECTED',

  LISTEN_REMOVED: 'LISTEN_REMOVED',

  REMOVE_LISTENER_REQUESTED: 'REMOVE_LISTENER_REQUESTED',
  REMOVE_ALL_LISTENERS_REQUESTED: 'REMOVE_ALL_LISTENERS_REQUESTED',

  CLEAR_ERROR: 'CLEAR_ERROR',
  CLEAR_AUTH_ERROR: 'CLEAR_AUTH_ERROR',

  SYNC_FLOCKS_REQUESTED: 'SYNC_FLOCKS_REQUESTED',
  SYNC_FLOCKS_REJECTED: 'SYNC_FLOCKS_REJECTED',
  SYNC_FLOCKS_FULFILLED: 'SYNC_FLOCKS_FULFILLED',

  SET_FLOCK: 'SET_FLOCK',
  CLEAR_FLOCK: 'CLEAR_FLOCK',
  CLEAR_ALL_FLOCKS: 'CLEAR_ALL_FLOCKS',

  UNLINK_FLOCK_REQUESTED: 'UNLINK_FLOCK_REQUESTED',
  UNLINK_FLOCK_FULFILLED: 'UNLINK_FLOCK_FULFILLED',

  DELETE_FLOCK_REQUESTED: 'DELETE_FLOCK_REQUESTED',
  DELETE_FLOCK_FULFILLED: 'DELETE_FLOCK_FULFILLED',
  DELETE_FLOCK_REJECTED: 'DELETE_FLOCK_REJECTED',

  DELETE_CHICKEN_REQUESTED: 'DELETE_CHICKEN_REQUESTED',
  DELETE_CHICKEN_FULFILLED: 'DELETE_CHICKEN_FULFILLED',
  DELETE_CHICKEN_REJECTED: 'DELETE_CHICKEN_REJECTED',

  JOIN_FLOCK_REQUESTED: 'JOIN_FLOCK_REQUESTED',
  JOIN_FLOCK_FULFILLED: 'JOIN_FLOCK_FULFILLED',
  JOIN_FLOCK_REJECTED: 'JOIN_FLOCK_REJECTED',

  ADD_FLOCK_REQUESTED: 'ADD_FLOCK_REQUESTED',
  ADD_FLOCK_FULFILLED: 'ADD_FLOCK_FULFILLED',
  ADD_FLOCK_REJECTED: 'ADD_FLOCK_REJECTED',

  SAVE_CHICKEN_REQUESTED: 'SAVE_CHICKEN_REQUESTED',

  CHILD_ADDED: 'CHILD_ADDED',
  CHILD_CHANGED: 'CHILD_CHANGED',
  CHILD_REMOVED: 'CHILD_REMOVED',

  SET_INITIAL_URL: 'SET_INITIAL_URL',
  REMOVE_INITIAL_URL: 'REMOVE_INITIAL_URL',

  AUTH_STATUS_LOGGED_IN: 'AUTH_STATUS_LOGGED_IN',
  AUTH_STATUS_LOGGED_OUT: 'AUTH_STATUS_LOGGED_OUT',

  SIGN_OUT_REQUESTED: 'SIGN_OUT_REQUESTED',

  AUTH_ACTION_REQUESTED: 'AUTH_ACTION_REQUESTED',
  AUTH_ACTION_FULFILLED: 'AUTH_ACTION_FULFILLED',
  AUTH_ACTION_REJECTED: 'AUTH_ACTION_REJECTED',

  STORAGE_DELETE_REQUESTED: 'STORAGE_DELETE_REQUESTED',
  STORAGE_DELETE_FULFILLED: 'STORAGE_DELETE_FULFILLED',
  STORAGE_DELETE_REJECTED: 'STORAGE_DELETE_REJECTED',

  STORAGE_UPLOAD_REQUESTED: 'STORAGE_UPLOAD_REQUESTED',
  STORAGE_UPLOAD_FULFILLED: 'STORAGE_UPLOAD_FULFILLED',
  STORAGE_UPLOAD_REJECTED: 'STORAGE_UPLOAD_REJECTED',
};

export const metaTypes = {
  userSettings: 'userSettings',
  flocks: 'flocks',
  chickens: 'chickens',
  eggs: 'eggs',
};

export const authTypes = {
  signUp: 'signUp',
  signIn: 'signIn',
  resetPassword: 'resetPassword',
};

export const eventTypes = {
  CHILD_ADDED_EVENT: 'CHILD_ADDED_EVENT',
  CHILD_REMOVED_EVENT: 'CHILD_REMOVED_EVENT',
  CHILD_CHANGED_EVENT: 'CHILD_CHANGED_EVENT',
  AUTH_STATUS_CHANGED: 'AUTH_STATUS',
};

export const appStates = {
  STARTING: 'STARTING',
  BUSY: 'BUSY',
  READY: 'READY',
};
