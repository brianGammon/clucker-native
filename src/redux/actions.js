import firebase from 'react-native-firebase';
import { metaTypes, actionTypes as a } from './constants';

export function firebaseListenRequested(ref, metaType) {
  return {
    type: a.LISTEN_REQUESTED,
    payload: { ref },
    meta: { type: metaType },
  };
}

export function firebaseListenRejected(error, metaType) {
  return {
    type: a.LISTEN_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseListenFulfilled({ key, value }, metaType) {
  return {
    type: a.LISTEN_FULFILLED,
    payload: { key, items: value },
    meta: { type: metaType },
  };
}

export function firebaseListenChildAdded(id, value, metaType) {
  return {
    type: a.LISTEN_CHILD_ADDED,
    payload: { id, value },
    meta: { type: metaType },
  };
}

export function firebaseListenChildChanged(id, value, metaType) {
  return {
    type: a.LISTEN_CHILD_CHANGED,
    payload: { id, value },
    meta: { type: metaType },
  };
}

export function firebaseListenChildRemoved(id, metaType) {
  return {
    type: a.LISTEN_CHILD_REMOVED,
    payload: { id },
    meta: { type: metaType },
  };
}

export function firebaseUpdateRequested(payload, metaType) {
  return {
    type: a.UPDATE_REQUESTED,
    payload,
    meta: { type: metaType },
  };
}

export function firebaseUpdateRejected(error, metaType) {
  return {
    type: a.UPDATE_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseUpdateFulfilled(metaType) {
  return {
    type: a.UPDATE_FULFILLED,
    payload: {},
    meta: { type: metaType },
  };
}

export function firebaseRemoveRequested(payload, metaType) {
  return {
    type: a.REMOVE_REQUESTED,
    payload,
    meta: { type: metaType },
  };
}

export function firebaseRemoveRejected(error, metaType) {
  return {
    type: a.REMOVE_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseRemoveFulfilled(metaType) {
  return {
    type: a.REMOVE_FULFILLED,
    payload: {},
    meta: { type: metaType },
  };
}

export function firebaseListenRemoved(clearItems, metaType) {
  return {
    type: a.LISTEN_REMOVED,
    payload: { clearItems },
    meta: { type: metaType },
  };
}

export function firebaseRemoveListenerRequested(clearItems, metaType) {
  return {
    type: a.REMOVE_LISTENER_REQUESTED,
    payload: { clearItems },
    meta: { type: metaType },
  };
}

export function firebaseRemoveAllListenersRequested() {
  return {
    type: a.REMOVE_ALL_LISTENERS_REQUESTED,
    payload: { clearItems: true },
  };
}

export function listenToChickens(flockId) {
  const ref = firebase.database().ref(`chickens/${flockId}`);
  return firebaseListenRequested(ref, metaTypes.chickens);
}

export function listenToEggs(flockId) {
  const ref = firebase.database().ref(`eggs/${flockId}`);
  return firebaseListenRequested(ref, metaTypes.eggs);
}

export function listenToUserSettings(uid) {
  const ref = firebase.database().ref(`userSettings/${uid}`);
  return firebaseListenRequested(ref, metaTypes.userSettings);
}

export function removeUserSettingsListenerRequested() {
  return firebaseRemoveListenerRequested(false, metaTypes.userSettings);
}

export function updateUserSettingsRequested(uid, displayName, currentFlockId, flocks) {
  return firebaseUpdateRequested(
    {
      uid,
      displayName,
      currentFlockId,
      flocks,
    },
    metaTypes.userSettings,
  );
}

export function removeUserSettingsRequested(uid) {
  return firebaseRemoveRequested({ uid }, metaTypes.userSettings);
}

export function getFlock(flockId) {
  const ref = firebase.database().ref(`flocks/${flockId}`);
  return {
    type: a.GET_FLOCK_REQUESTED,
    payload: { ref },
    meta: { type: metaTypes.flocks },
  };
}

export function getFlockRejected(error) {
  return {
    type: a.GET_FLOCK_REJECTED,
    payload: { error },
    meta: { type: metaTypes.flocks },
  };
}

export function getFlockFulfilled(flock) {
  return {
    type: a.GET_FLOCK_FULFILLED,
    payload: flock,
    meta: { type: metaTypes.flocks },
  };
}

export function setInitialUrl(url) {
  return {
    type: a.SET_INITIAL_URL,
    payload: url,
  };
}

export function removeInitialUrl() {
  return {
    type: a.REMOVE_INITIAL_URL,
  };
}
