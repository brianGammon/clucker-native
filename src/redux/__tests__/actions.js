import firebase from 'react-native-firebase';
import * as actions from '../actions';
import * as constants from '../constants';

describe('Redux actions:', () => {
  test(`${constants.actionTypes.LISTEN_REQUESTED}`, () => {
    const ref = firebase.database().ref('someRef');
    expect(
      actions.firebaseListenRequested(ref, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.LISTEN_REJECTED}`, () => {
    const error = new Error('Error!');
    expect(
      actions.firebaseListenRejected(error, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.LISTEN_FULFILLED}`, () => {
    const data = { item1: 1, item2: 2 };
    expect(
      actions.firebaseListenFulfilled(data, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.CHILD_ADDED}`, () => {
    const id = '1';
    const value = 'a';

    expect(
      actions.firebaseListenChildAdded(id, value, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.CHILD_CHANGED}`, () => {
    const id = '1';
    const value = 'a';

    expect(
      actions.firebaseListenChildChanged(id, value, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.CHILD_REMOVED}`, () => {
    const id = '1';
    expect(
      actions.firebaseListenChildRemoved(id, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.CREATE_REQUESTED}`, () => {
    const payload = {
      flockId: 'flock1',
      data: {
        name: 'Test Chicken',
      },
    };
    expect(
      actions.firebaseCreateRequested(payload, 'chickens'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.CREATE_FULFILLED}`, () => {
    expect(actions.firebaseCreateFulfilled('chickens')).toMatchSnapshot();
  });

  test(`${constants.actionTypes.CREATE_REJECTED}`, () => {
    const error = new Error('Error!');

    expect(actions.firebaseCreateRejected(error, 'chickens')).toMatchSnapshot();
  });

  test(`${constants.actionTypes.UPDATE_REQUESTED}`, () => {
    const uid = '1';

    expect(
      actions.firebaseUpdateRequested(uid, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.UPDATE_FULFILLED}`, () => {
    expect(actions.firebaseUpdateFulfilled('userContacts')).toMatchSnapshot();
  });

  test(`${constants.actionTypes.UPDATE_REJECTED}`, () => {
    const error = new Error('Error!');

    expect(
      actions.firebaseUpdateRejected(error, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.REMOVE_REQUESTED}`, () => {
    const uid = '1';

    expect(
      actions.firebaseRemoveRequested(uid, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.REMOVE_FULFILLED}`, () => {
    expect(actions.firebaseRemoveFulfilled('userContacts')).toMatchSnapshot();
  });

  test(`${constants.actionTypes.REMOVE_REJECTED}`, () => {
    const error = new Error('Error!');

    expect(
      actions.firebaseRemoveRejected(error, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.LISTEN_REMOVED}`, () => {
    expect(
      actions.firebaseListenRemoved(true, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.REMOVE_LISTENER_REQUESTED}`, () => {
    expect(
      actions.firebaseRemoveListenerRequested(true, 'userContacts'),
    ).toMatchSnapshot();
  });

  test(`${constants.actionTypes.REMOVE_ALL_LISTENERS_REQUESTED}`, () => {
    expect(actions.firebaseRemoveAllListenersRequested()).toMatchSnapshot();
  });

  test('listenToChickens', () => {
    expect(actions.listenToChickens('user1')).toMatchSnapshot();
  });

  test('listenToEggs', () => {
    expect(actions.listenToEggs('user1')).toMatchSnapshot();
  });

  test('setInitialUrl', () => {
    expect(actions.setInitialUrl('test://test')).toMatchSnapshot();
  });

  test('removeInitialUrl', () => {
    expect(actions.removeInitialUrl()).toMatchSnapshot();
  });

  test('authStatusChanged to authenticated', () => {
    const user = { uid: 'test', email: 'test@example.com' };
    expect(actions.authStatusChanged(user)).toMatchSnapshot();
  });

  test('authStatusChanged to not authenticated', () => {
    expect(actions.authStatusChanged(null)).toMatchSnapshot();
  });

  test('signInRequested', () => {
    expect(
      actions.signInRequested('test@example.com', 'password123'),
    ).toMatchSnapshot();
  });

  test('signUpRequested', () => {
    expect(
      actions.signUpRequested('test@example.com', 'password123'),
    ).toMatchSnapshot();
  });

  test('resetPasswordRequested', () => {
    expect(
      actions.resetPasswordRequested('test@example.com'),
    ).toMatchSnapshot();
  });
});
