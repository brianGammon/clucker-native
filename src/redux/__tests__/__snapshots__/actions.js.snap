// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux actions: LISTEN_CHILD_ADDED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "id": "1",
    "value": "a",
  },
  "type": "LISTEN_CHILD_ADDED",
}
`;

exports[`Redux actions: LISTEN_CHILD_CHANGED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "id": "1",
    "value": "a",
  },
  "type": "LISTEN_CHILD_CHANGED",
}
`;

exports[`Redux actions: LISTEN_CHILD_REMOVED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "id": "1",
  },
  "type": "LISTEN_CHILD_REMOVED",
}
`;

exports[`Redux actions: LISTEN_FULFILLED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "items": Object {
      "item1": 1,
      "item2": 2,
    },
    "key": "someKey",
  },
  "type": "LISTEN_FULFILLED",
}
`;

exports[`Redux actions: LISTEN_REJECTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "error": [Error: Error!],
  },
  "type": "LISTEN_REJECTED",
}
`;

exports[`Redux actions: LISTEN_REMOVED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "clearItems": true,
  },
  "type": "LISTEN_REMOVED",
}
`;

exports[`Redux actions: LISTEN_REQUESTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "ref": Reference {
      "data": null,
      "off": [MockFunction],
      "on": [MockFunction],
      "once": [MockFunction],
      "path": "someRef",
      "remove": [MockFunction],
      "snap": Object {
        "val": [Function],
      },
      "update": [MockFunction],
      "val": [MockFunction],
    },
  },
  "type": "LISTEN_REQUESTED",
}
`;

exports[`Redux actions: REMOVE_ALL_LISTENERS_REQUESTED 1`] = `
Object {
  "payload": Object {
    "clearItems": true,
  },
  "type": "REMOVE_ALL_LISTENERS_REQUESTED",
}
`;

exports[`Redux actions: REMOVE_FULFILLED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {},
  "type": "REMOVE_FULFILLED",
}
`;

exports[`Redux actions: REMOVE_LISTENER_REQUESTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "clearItems": true,
  },
  "type": "REMOVE_LISTENER_REQUESTED",
}
`;

exports[`Redux actions: REMOVE_REJECTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "error": [Error: Error!],
  },
  "type": "REMOVE_REJECTED",
}
`;

exports[`Redux actions: REMOVE_REQUESTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": "1",
  "type": "REMOVE_REQUESTED",
}
`;

exports[`Redux actions: UPDATE_FULFILLED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {},
  "type": "UPDATE_FULFILLED",
}
`;

exports[`Redux actions: UPDATE_REJECTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": Object {
    "error": [Error: Error!],
  },
  "type": "UPDATE_REJECTED",
}
`;

exports[`Redux actions: UPDATE_REQUESTED 1`] = `
Object {
  "meta": Object {
    "type": "userContacts",
  },
  "payload": "1",
  "type": "UPDATE_REQUESTED",
}
`;

exports[`Redux actions: getFlock 1`] = `
Object {
  "meta": Object {
    "type": "flocks",
  },
  "payload": Object {
    "ref": Reference {
      "data": null,
      "off": [MockFunction],
      "on": [MockFunction],
      "once": [MockFunction],
      "path": "flocks/flockId1",
      "remove": [MockFunction],
      "snap": Object {
        "val": [Function],
      },
      "update": [MockFunction],
      "val": [MockFunction],
    },
  },
  "type": "GET_FLOCK_REQUESTED",
}
`;

exports[`Redux actions: getFlockFulfilled 1`] = `
Object {
  "meta": Object {
    "type": "flocks",
  },
  "payload": Object {
    "name": "Test Flock",
    "ownedBy": "user1",
  },
  "type": "GET_FLOCK_FULFILLED",
}
`;

exports[`Redux actions: getFlockRejected 1`] = `
Object {
  "meta": Object {
    "type": "flocks",
  },
  "payload": Object {
    "error": [Error: Error1],
  },
  "type": "GET_FLOCK_REJECTED",
}
`;

exports[`Redux actions: listenToChickens 1`] = `
Object {
  "meta": Object {
    "type": "chickens",
  },
  "payload": Object {
    "ref": Reference {
      "data": null,
      "off": [MockFunction],
      "on": [MockFunction],
      "once": [MockFunction],
      "path": "chickens/flockId1",
      "remove": [MockFunction],
      "snap": Object {
        "val": [Function],
      },
      "update": [MockFunction],
      "val": [MockFunction],
    },
  },
  "type": "LISTEN_REQUESTED",
}
`;

exports[`Redux actions: listenToEggs 1`] = `
Object {
  "meta": Object {
    "type": "eggs",
  },
  "payload": Object {
    "ref": Reference {
      "data": null,
      "off": [MockFunction],
      "on": [MockFunction],
      "once": [MockFunction],
      "path": "eggs/flockId1",
      "remove": [MockFunction],
      "snap": Object {
        "val": [Function],
      },
      "update": [MockFunction],
      "val": [MockFunction],
    },
  },
  "type": "LISTEN_REQUESTED",
}
`;

exports[`Redux actions: listenToUserSettings 1`] = `
Object {
  "meta": Object {
    "type": "userSettings",
  },
  "payload": Object {
    "ref": Reference {
      "data": null,
      "off": [MockFunction],
      "on": [MockFunction],
      "once": [MockFunction],
      "path": "userSettings/userId1",
      "remove": [MockFunction],
      "snap": Object {
        "val": [Function],
      },
      "update": [MockFunction],
      "val": [MockFunction],
    },
  },
  "type": "LISTEN_REQUESTED",
}
`;

exports[`Redux actions: removeInitialUrl 1`] = `
Object {
  "type": "REMOVE_INITIAL_URL",
}
`;

exports[`Redux actions: removeUserSettingsListenerRequested 1`] = `
Object {
  "meta": Object {
    "type": "userSettings",
  },
  "payload": Object {
    "clearItems": false,
  },
  "type": "REMOVE_LISTENER_REQUESTED",
}
`;

exports[`Redux actions: removeUserSettingsRequested 1`] = `
Object {
  "meta": Object {
    "type": "userSettings",
  },
  "payload": Object {
    "uid": "userId1",
  },
  "type": "REMOVE_REQUESTED",
}
`;

exports[`Redux actions: setInitialUrl 1`] = `
Object {
  "payload": "test://test",
  "type": "SET_INITIAL_URL",
}
`;

exports[`Redux actions: updateUserSettingsRequested 1`] = `
Object {
  "meta": Object {
    "type": "userSettings",
  },
  "payload": Object {
    "currentFlockId": "flockId1",
    "displayName": "some name",
    "flocks": Object {
      "flockId1": true,
    },
    "uid": "userId1",
  },
  "type": "UPDATE_REQUESTED",
}
`;
